### ISSUE TITLE
Karabiner elements breaks the TouchBar on the new MacBook pros

### ISSUE THREAD

**Issue by DJCool1** [tekezo/Karabiner-Elements#435]
None of my touchbar buttons worked after installing Karabiner elements. After uninstalling it, they all immediately started working again.

**Comment by kyleholzinger** [comment]
+1

**Comment by kyleholzinger** [comment]
@DJCool1 [check out this one](https://github.com/tekezo/Karabiner-Elements/issues/431). Decent fix, makes sense since I guess macs think that it's another keyboard.

**Comment by DJCool1** [comment]
Ah thanks, that fixed it. For some reason, in my case there was not just *one* but *two* "No product name" keyboards listed, so I unchecked both.

**Comment by Dgr8Filippo** [comment]
I had issues too - strangely enough the only FN keys affected were Mission Control's F11 and F12 for show desktop and dashboard respectively.  They were "stuck" at volume down and volume up instead.  I uninstalled Karabiner Elements and I got my keys back.  Incidentally this was with the latest version (as of this writing).

**Comment by broberts01** [comment]
Yup. I'm experiencing this too.   The unchecking "No Product Name" solved the Mission Control/KB/iTunes issue, but as noted the holding down Fn is still broken.

Thanks!

**Comment by allenmhc** [comment]
Just ran into the same issues as others here. To be specific, though, the Touch Bar buttons that fail seem to be the ones formerly mapped to the Fn keys: mute, volume up/down (but *not* the volume button that goes into the slider), play/pause (but not when in iTunes), Mission Control, keyboard lighting, etc. It's probably just incorrectly intercepting those events.

**Comment by broberts01** [comment]
Another important one that no longer works when Elements is installed is Option-Command-Esc to Force Quit.

**Comment by kyleholzinger** [comment]
Good point @allenmhc ! I think you're right. Interesting that it is registered as a keyboard. @broberts01 that's weird, so holding down the fn key doesn't show up the function buttons? For me it does, and f11 works as "show desktop"

**Comment by broberts01** [comment]
@kyleholzinger:  Correct w/o Elements holding Fn shows F1 through F12 in any app.  With Elements Fn has no effect on the Touch Bar (though if you hit Fn twice dictation will start as expected).

**Comment by kyleholzinger** [comment]
Hm interesting. @broberts01 so that's even after you turned off the "No Product Name" input in the devices preferences?

**Comment by broberts01** [comment]
@kyleholzinger: Correct. The "No Product Name" is unchecked.

**Comment by noestreich** [comment]
i can attest to the observations @broberts01 is describing. I have the same exact issue. 
Karabiner is installed, the ominous box unchecked. Volume- and Mediaplayer-Buttons work again, but the FN-Key is not able to bringt up the function-key-row. Without Elements, holding FN shows F1 through F12 in any app - with Elements FN has no effect on the Touch Bar.

**Comment by kyleholzinger** [comment]
I wonder if you uninstall karabiner if it work @noestreich @broberts01

**Comment by noestreich** [comment]
@kyleholzinger funny, I just read your reply on my phone opened up the MacBook and wanted to reply:"Yes it works when Karabiner is uninstalled but not as described above", but then i realized, that the FN key worked again. This was the first time my MacBook went to sleep since my last reinstall of Karabiner maybe there is a connection.

**Comment by kyleholzinger** [comment]
Hm interesting @noestreich. No idea! I've never used karabiner with sierra so no idea. Glad it works now though! Maybe that's the answer

**Comment by vincent99** [comment]
I have the same problem, with Karabiner Elements installed and "No product name" disabled the `fn` key on the MacBook has no effect on the Touch Bar.

Interestingly, if I add an external Apple keyboard (0x024f) *and* disable modifying events for it, then the `fn` key on the internal keyboard still does nothing but `fn` on the external one **does** toggle the Touch Bar correctly.

**Comment by kyleholzinger** [comment]
@vincent99 I suspect something else is going on, seems like the touchbar is definitely another input device. I'd check your installed programs for seil or karabiner or something, but for everyone else it seems like for the most part disabling the "no product name" device solved this issue

**Comment by broberts01** [comment]
Just so we're clear -- disabling the "no product name" device solves the Mission Control/KB/iTunes issue, but as noted the holding down Fn to get the function keys is still broken as is any KB combination that includes the Fn keys or Esc.  e.g Cmd-Optn-Esc, Cmd-F9, etc.

**Comment by wjohnson191** [comment]
When Karabiner-Elements 0.90.67 is running, holding down the fn key doesn't cause the function keys to appear on the Touch Bar. When I quit Karabiner-Elements, the problem goes away. Under "Devices", "No Product Name" shows up twice, both of which are unselected (and were unselected by default after installation). Under "Function Keys", "Use all F1..." can either be selected or deselected, the problem occurs either way.

**Comment by DannyVenezuela** [comment]
I have noticed EventViewer registers the keystrokes for mute, next track, previous track, play/pause, volume up, volume down as "SysKeyUp" eventTypes, but never a "SysKeyDown" on the TouchBar. Esc, keyboard brightness, and screen brightness all register as "SysKeyDown." I don't know enough to fix the issue but it seemed important.

**Comment by zagrebin** [comment]
+1

**Comment by SomeoneToIgnore** [comment]
I would also like to notice that for all applications you force the function keys to appear (System Preferences -> Keyboard -> Shortcuts -> Function Keys menu), those keys do not work with any kind of modifier key: Shift, Ctrl, Alt or Cmd + any function key is treated as function key only.

**Comment by eddyg** [comment]
Any updates on resolving these issues for MacBook Pro TouchBar users?

1. `fn` key does not toggle TouchBar mode display
2. `⌘+⌥+⎋` (command+option+esc) to open Force Quit is ignored

**Comment by mattPiratt** [comment]
I am experiencing the same problem with karabinier elements and the Fn key. I would love to see a fix or feedback

**Comment by samfearn** [comment]
I'm having the same problem as others; if I disable the 'no product name' device, then the fn key works to bring up the extended touchbar (with it enabled, this doesn't work), but disabling this device stops the escape key from functioning properly, this effects force-quit as well as other shortcuts involving escape. With the device disabled, escape does work as a stand-alone key, just not with modifier keys.

**Comment by neohunter** [comment]
Same problem here ,FN dont display the FN keys on touchbar

**Comment by nicklasbondesson** [comment]
Any progress here?

**Comment by nivshah** [comment]
I've had good luck disabling the "No product name" device from Karabiner Elements as a workaround.

**Comment by makslevental** [comment]
+1 for FN doesn't do anything when karabiner running

**Comment by kyleholzinger** [comment]
@makslevental you've tried the above suggestions of disabling the "no product name" as a device?

**Comment by makslevental** [comment]
@kyleholzinger correct

**Comment by iansinnott** [comment]
@makslevental what others have suggested above regarding letting the computer sleep worked for me. Put it to sleep or go into screensaver then log back in. Fn keys work again... Although this is not ideal, it might partially solve the issue.

**Comment by caepenna** [comment]
I have had similar issues (the first few keys of my control stripe stopped working after installing karabiner elements) but I can see no way to fix it, even after uninstalling karabiner and rebooting the system.

And there are people with the same issue on [Apple's forum](https://discussions.apple.com/message/31432265#message31432265).

**Comment by simspace-dr** [comment]
Any progress on the issue that prevents Option-Command-Esc from opening Force Quit when Elements running? Whenever I want to use Option-Command-Esc to open Force Quit, I have to first quit Elements.

**Comment by simspace-dr** [comment]
Here's a link to my screen recording on YouTube showing the  Option-Command-Esc bug: https://youtu.be/XxQ745Zhz6k

**Comment by cotfas** [comment]
I have the same issue

**Comment by r00ger** [comment]
Yaah, me too

**Comment by chinchin96** [comment]
I'm experiencing the same thing.  I can't open the Force Quit dialog window (Option-Command-Esc) when "No Product Name" is unchecked.  However, if I check it, my display brightness, keyboard backlighting controls don't work.  

It sounds like with some rewiring this can be fixed where it's no longer mutually exclusive. Has anyone found a workaround other than checking/unchecking the boxes?

**Comment by backslash112** [comment]
#535

**Comment by baharestani** [comment]
There is a good solution [here](https://github.com/tekezo/Karabiner-Elements/issues/535).
You have to apply the one from @mutantcornholio and @kasecato

**Comment by jkldgoefgkljefogeg** [comment]
Karabiner-Elements
  * fn to left_command
  * left_command to left_control
  * left_control to fn
  * checked Modifying events from No product name
  * Use all F1, F2 keys as standard function keys

MacOS
  * Touchbar shows F1/F2
  * Press Fn to Expand Control Strip

Still Touchbar Expand Control Strip when I press either physical fn or left_control key, in my case making it impossible to perform left_command (physical Fn key) + F1

**Comment by studgeek** [comment]
To save folks having to dig through the entire other thread, the workaround on Karabiner side is this post (enable keybaord 1452/34304) https://github.com/tekezo/Karabiner-Elements/issues/535#issuecomment-350522019 plus possibly this post https://github.com/tekezo/Karabiner-Elements/issues/535#issuecomment-350597725.

**Comment by studgeek** [comment]
I also filed #1700 as a feature request to have that keyboard enabled by default.

**Comment by psillypsymun** [comment]
Well this sucks.  I installed Karabiner and noticed my Touch Bar wasn't working.  So I uninstalled it and it's still not lighting up.  I just got this MacBook today too.  Was there a a warning I missed when I installed it? Because that would have been nice.
